### PR TITLE
Improve zigcc support

### PR DESCRIPTION
**Pull request by Redbeanw44602**
- fix ar
 - fix cmake build (add wrapper)

https://github.com/xmake-io/xmake/issues/5610

### CODE REVIEW

File: xmake/toolchains/zig/xmake.lua [hunk 1]
```diff
@@ -52,24 +52,43 @@ toolchain("zig")
     on_load(function (toolchain)
         import("core.base.semver")
 
+        -- @see https://github.com/xmake-io/xmake/issues/5610
+        function _generate_zigcc_wrapper(zig)
+            local tools = {}
+            for _, tool in ipairs { "cc", "c++", "ar", "ranlib", "objcopy" } do
```
Comment (waruqi) [inline xmake/toolchains/zig/xmake.lua:58]: add `()`, `ipairs({"cc", "c++", "ar", "ranlib", "objcopy"})`

File: xmake/toolchains/zig/xmake.lua [hunk 2]
```diff
@@ -52,24 +52,43 @@ toolchain("zig")
...
+                local wrapper_path = path.join(os.tmpdir(), "zigcc", tool)
+                if not os.is_host("windows") then
```
Comment (waruqi) [inline xmake/toolchains/zig/xmake.lua:60]: `os.is_host` -> is_host

File: xmake/toolchains/zig/xmake.lua [hunk 3]
```diff
@@ -52,24 +52,43 @@ toolchain("zig")
     on_load(function (toolchain)
         import("core.base.semver")
 
+        -- @see https://github.com/xmake-io/xmake/issues/5610
+        function _generate_zigcc_wrapper(zig)
```
Comment (waruqi) [inline xmake/toolchains/zig/xmake.lua:56]: we should wrap these tools in toolchain/on_check, and save to toolchain:config(), then get it in on_load

**Comment by waruqi** [comment]
it does not work on macOS. but it works for dev branch

```
/var/folders/32/w9cz0y_14hs19lkbs6v6_fm80000gn/T/.xmake501/241203/zigcc/cc -c -Qunused-argument
s -target x86_64-apple-macos14.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms
/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -fvisibility=hidden -DNDEBUG -o build/.objs/test
/macosx/x86_64/release/src/main.c.o src/main.c
checking for flags (-MMD -MF) ... ok
checking for flags (-fdiagnostics-color=always) ... no
checking for flags (-fcolor-diagnostics) ... ok
checking for flags (-Wno-gnu-line-marker -Werror) ... no
error: error: unable to parse target query 'x86_64-apple-macos14.0': UnknownOperatingSystem
```

**Comment by Redbeanw44602** [comment]
> it does not work on macOS. but it works for dev branch
> 
> ```
> /var/folders/32/w9cz0y_14hs19lkbs6v6_fm80000gn/T/.xmake501/241203/zigcc/cc -c -Qunused-argument
> s -target x86_64-apple-macos14.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms
> /MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -fvisibility=hidden -DNDEBUG -o build/.objs/test
> /macosx/x86_64/release/src/main.c.o src/main.c
> checking for flags (-MMD -MF) ... ok
> checking for flags (-fdiagnostics-color=always) ... no
> checking for flags (-fcolor-diagnostics) ... ok
> checking for flags (-Wno-gnu-line-marker -Werror) ... no
> error: error: unable to parse target query 'x86_64-apple-macos14.0': UnknownOperatingSystem
> ```

Sorry I don't have a macos device, according to https://github.com/Nuitka/Nuitka/issues/2756, maybe `apple-` is unnecessary?

**Comment by Redbeanw44602** [comment]
```
$ zig c++ -target x86_64-macos simple.cpp
$ file a.out
a.out: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|NO_REEXPORTED_DYLIBS|PIE>
```
It seems that the correct target should be `x86_64-macos`.

**Comment by waruqi** [comment]
> ```
> $ zig c++ -target x86_64-macos simple.cpp
> $ file a.out
> a.out: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|NO_REEXPORTED_DYLIBS|PIE>
> ```
> 
> It seems that the correct target should be `x86_64-macos`(-none).

if I do not apply this patch, it will use `-target x86_64-macos` on dev branch.

**Comment by Redbeanw44602** [comment]
> if I do not apply this patch, it will use `-target x86_64-macos` on dev branch.

I see, I'm checking what went wrong...

**Comment by waruqi** [comment]
it will add target flags for zig cc/c++

https://github.com/xmake-io/xmake/blob/b3c31167540804204325709f40eb1eb0f8b52b18/xmake/toolchains/zig/xmake.lua#L123-L129

but its name has been rename to `cc`

**Comment by Redbeanw44602** [comment]
I didn't see any renames happening in the diff, and if that's the problem, it shouldn't just affect macOS (it works on my linux🤔)

**Comment by waruqi** [comment]
> I didn't see any renames happening in the diff, and if that's the problem, it shouldn't just affect macOS (it works on my linux🤔)

if you rename to cc/c++, it will use `tools/cc.lua` instead of `tools/zig_cc.lua`, zig_cc.lua will do some thing for zig cc

**Pull request merged**